src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle {
+  shape: Shape = 'triangle';
 
-}
+  color: Color;
+
+  a: number;
+
+  b: number;
 
-export class Circle {
+  c: number;
 
+  constructor(color: Color, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Sides cannot form a triangle');
+    }
+
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    return Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  color: Color;
+
+  radius: number;
 
+  constructor(color: Color, radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    return Math.PI * this.radius * this.radius;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: Shape = 'rectangle';
+
+  color: Color;
+
+  width: number;
+
+  height: number;
```

Comment:
a lot of spaces, it can be simplify 
```suggestion
  shape: Shape = 'rectangle';
  color: Color;

  width: number;
  height: number;
```

Reply:
I`m add a '@typescript-eslint/lines-between-class-members': 'off' to .eslintrc.js